components/vocabularies/omim/api/src/main/java/org/phenotips/vocabulary/internal/solr/MendelianInheritanceInMan.java

```diff
@@ -243,4 +338,183 @@ private int clear()
         }
         return 1;
     }
+
+    private void index()
+    {
+        try {
+            parseOmimData();
+            loadGenes();
+            loadSymptoms(true);
+            loadSymptoms(false);
+            loadGeneReviews();
+            loadVersion();
+        } catch (NullPointerException ex) {
+            this.logger.error("Failed to prepare the OMIM index: {}", ex.getMessage(), ex);
+        }
+    }
+
+    private void parseOmimData()
+    {
+        try {
+            Reader in =
+                new InputStreamReader(new URL(OMIM_SOURCE_URL).openConnection().getInputStream(),
+                    Charset.forName(ENCODING));
+            for (CSVRecord row : CSVFormat.TDF.parse(in)) {
+                // ignore the comments, moved or removed entries
+                if (row.get(0).startsWith("#") || ("Caret").equals(row.get(0))
```

Comment:
Instead of manually checking for `#` each time, you can just use `CSVFormat.TDF.withCommentMarker('#')` to ignore comment lines automatically.